index.html

```diff
@@ -4677,98 +4722,165 @@ <h2>Internationalization Considerations</h2>
+developers that primarily utilize JSON and CBOR to publish information. The
+pattern is useful for the vast majority of language and base direction use
+cases that implementers of this specification will need to support.
       </p>
 
-      <ul>
-        <li>
-[[!JSON-LD]] supports the expression of language information using the
-<code>@language</code> feature, as described in [[!JSON-LD]] 1.1,
-<a href="https://www.w3.org/TR/json-ld11/#string-internationalization">Section 4.2.3: String Internationalization</a>.
-        </li>
-        <li>
-[[!JSON-LD]] also supports the expression of language information using the
-<code>rdf:HTML</code> string literal type, as described in [[!JSON-LD]] 1.0,
-<a href="https://www.w3.org/TR/json-ld11/#typed-values">Section 4.2.1: Typed Values</a>.
-This mechanism enables developers to use the internationalization features
-present in HTML to express language information. For example, consider the
-following key-value pair:
+      <p>
+The general design pattern is to use the following markup template when
+expressing a text string that is tagged with a language and, optionally, a
+specific text direction:
+      </p>
 
-          <pre>
-"nameHtml": "&lt;span lang="jp">和美&lt;/span>"
-          </pre>
+      <pre class="example nohighlight" title="Design pattern for natural language strings">
+"<a>property</a>": {
+  "value": "<span class="highlight">The string value</span>",
+  "language: "<code>LANGUAGE</code>-x-dir-<code>DIRECTION</code>"
+}
+      </pre>
 
-          <p>
-If the JSON-LD Context expressed that any value associated with the
-<code>nameHtml</code> property as being of type <code>rdf:HTML</code>, then
-software agents rendering the property can deterministically identify the
-language in use.
-          </p>
-        </li>
-      </ul>
-    </section>
+      <p>
+An example utilizing the design pattern above is shown below to express the
+human-readable text "revoked" in the English language without specifying a
+text direction:
+      </p>
 
-    <section class="informative">
-      <h3>Text direction</h3>
+      <pre class="example nohighlight" title="Expressing natural language text as English">
+"statusReason": {
+  "value": "<span class="highlight">Revoked</span>",
+  "language: "<code>en</code>"
+}
+      </pre>
 
       <p>
-When expressing human-readable text content in the data model, it is important
-to be able to explicitly encode the text direction. JSON and its derived formats
-use the UTF-8 encoding of [[Unicode]] for serializing and transmitting text.
-This enables text direction to be directly determined in simple cases,
-but the mechanism is insufficient for general use.
+The next example expresses the human-readable text "revoked" in the Arabic
+language with a base direction of right-to-left:
       </p>
 
+      <pre class="example nohighlight" title="Arabic text with a base direction of right-to-left">
```

Comment:
My understanding (which has changed many times on this issue, so I wouldn't be surprised if it needs more updating)... is that the primary reason for adding "direction" is to handle cases where the language tag is *incorrect*  or missing but there is direction meta data that *is* correct. There is apparently a common pattern where the wrong language tag gets assigned to some data but the direction is proper and should be given preference. Because of this situation, "direction" is being added to capture this otherwise correct and dropped meta data.

For example, the text is Arabic, it is marked with an incorrect tag of "en" but the "direction" is properly "rtl".

I realize it would be odd to add an example with this erroneous information, but isn't that the point of the feature? We don't need it otherwise.

Reply:
Chaals is correct to point out that the direction information is not essential for a single arabic word, such as used in the example currently.

If the word was "Revoked!"  it _would_ be necessary for the consumer to apply RTL base direction, so that the exclamation mark appears on the correct side of the arabic word. However, if the consumer is using first-strong heuristics to detect direction (which we recommend) then it can determine the direction from that, since the first letter in the arabic word is RTL.

If, however, the first letter in an arabic translation is not RTL, then the base direction needs to be indicated, since otherwise the heuristics will make the wrong decision.  So an example with some initial text in latin script would be a stronger example.

In string-meta we use:
HTML و CSS: تصميم و إنشاء مواقع الويب
which means 
HTML and CSS: Designing and Creating Websites

Btw, there's another issue which is common when dealing with examples of bidi text written in english: How do you show the arabic text in the example?  The way a user would see if correctly displayed is with "CSS و HTML" to the right of the remaining arabic text.  To achieve that in the example, you'd need to use markup under the hood.  However, then it may be less clear to the casual ready of the spec that the string begins with H.  It's not a problem with an easy solution.  Especially in Arabic, where overriding the text to show order of characters in memory produces strange joining behaviour (which is why we use Hebrew for that).

Reply:
I slightly disagree with @r12a here. The base direction is needed for other reasons besides just the bidi algorithm. For example, layout, text alignment, or bullet placement in a list might depend on the base direction. Sometimes we need base direction information even when it doesn't affect the text composition.

It's of course helpful to choose examples in which setting the direction has some observable effect, and this always means mixed direction text. The best examples start (as ours does) with a strongly "anti-directional" character and ends (as ours does not) with punctuation (`?` is ideal); even better if one can get a number and some enclosing punctuation (parentheses).

Reply:
@aphillips, if you're referring to the first para in my comment about a single arabic word, i agree that you'd need an overall base direction to be applied by the consumer if it wasn't inserted into some other text, but the  FS heuristics would get you that base direction.

Btw, a caution about ?, since Arabic text would use ؟ U+061F ARABIC QUESTION MARK, which has the bidi category of AL - Right-to-left Arabic.  In short, it will end up in the right place when dropped into a LTR context if it follows arabic letters. It could be used with Hebrew examples, however, since Hebrew uses the european question mark.